modules/battery_manager/ajax/add_or_edit_entry.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * Battery Manager entry adder and editer
+ *
+ * Checks Test Battery for duplicates
+ * Handles insertions and updates in the Test Battery
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Battery_Manager
+ * @author   Victoria Foing <victoria.foing@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+
+// Determine which action is called
+if (isset($_GET['action'])) {
+    $action = $_GET['action'];
+    if ($action == "checkForDuplicate") {
+        echo checkForDuplicate();
+    } else if ($action == "add") {
+        addEntry();
+    } else if ($action == "edit") {
+        editEntry();
+    }
+}
+
+/**
+ * Return duplicate of entry if it exists and null otherwise
+ *
+ * @return json object or null
+ */
+function checkForDuplicate()
```

Comment:
please include return type in function definition

Reply:
also, the return type appears to be either null or string, not a json object.